shell/run_shell_cmd.go

```diff
@@ -42,17 +42,36 @@ func RunShellCommandWithOutput(terragruntOptions *options.TerragruntOptions, wor
 	var stdoutBuf bytes.Buffer
 	var stderrBuf bytes.Buffer
 
-	cmd := exec.Command(command, args...)
+	/**
+	 * perform Interpolation on arguments, copy the result to new array and pass this
+	 * to exec.Commnand
+	 */
+	include := terragruntOptions.GetIncludeConfig()
+	argsParsed := make([]string, len(args))
+	for i := 0; i < len(args); i++ {
+		value, err := ResolveTerragruntConfigString(args[i], include, terragruntOptions)
+		if err != nil {
+			return nil, errors.WithStackTrace(err)
+		}
+		argsParsed[i] = value
+	}
```

Comment:
This again feels like an abstraction leak... I understand you're trying to do the binding just before we execute the command, but `RunTerraformCommandWithOutput` is a general purpose method that should know nothing of interpolations. I think interpolation processing needs to happen earlier, in some wrapper method.

Reply:
Okay ... no I understood your two complains here .. I was a bit confused about what you meant about the 'abstraction leak'. As I said when I filed the PR we changed the code of terraform in the customer project with the most minimal effort possible. There was not much time to think about abstraction layers.